### DISCUSSION TITLE
doris1.0.0 Unique模型 + 开启向量化执行出现内存泄漏

### DISCUSSION THREAD

**Discussion by 980150976** [apache/doris#10371] [Show and tell]
表模型：unique 

`CREATE TABLE `pack_test` (
  `start_time` bigint(20) NULL COMMENT "开始采集时间",
  `end_time` bigint(20) NULL COMMENT "结束采集时间",
  `_id` varchar(255) NULL COMMENT "_id",
  `dt` date NULL COMMENT "日期，例如2022-05-10",
  `artifact_id` int(11) NULL COMMENT "artifact_id",
   xx
) ENGINE=OLAP
UNIQUE KEY(`start_time`, `end_time`, `_id`, `dt`)
COMMENT "OLAP"
PARTITION BY RANGE(`dt`)
(PARTITION p2019 VALUES [('0000-01-01'), ('2020-01-01')),
PARTITION p2020 VALUES [('2020-01-01'), ('2021-01-01')),
PARTITION p2021 VALUES [('2021-01-01'), ('2022-01-01')),
PARTITION p2022 VALUES [('2022-01-01'), ('2023-01-01')))
DISTRIBUTED BY HASH(`_id`) BUCKETS 16
PROPERTIES (
"replication_allocation" = "tag.location.default: 3",
"bloom_filter_columns" = "duid, puid, name, _id, artifact_id, url",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "month",
"dynamic_partition.time_zone" = "Asia/Shanghai",
"dynamic_partition.start" = "-2147483648",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.replication_allocation" = "tag.location.default: 3",
"dynamic_partition.buckets" = "16",
"dynamic_partition.create_history_partition" = "false",
"dynamic_partition.history_partition_num" = "-1",
"dynamic_partition.hot_partition_num" = "0",
"dynamic_partition.reserved_history_periods" = "NULL",
"dynamic_partition.start_day_of_month" = "1",
"in_memory" = "false",
"storage_format" = "V2"
)`

表总数据量1000w左右，占空间11G左右

set global enable_vectorized_engine = true
set global  batch_size = 4096;

集群内存>1T
![内存泄漏](https://user-images.githubusercontent.com/9603864/175264359-9cf0d6b4-a5fa-4362-99b8-96f585e82cf8.jpg)

不断查询后，内存持续飙升，直到BE挂掉

尝试用明细模型，没有这个问题！！！

**Comment by 980150976** [comment]
doris版本：https://www.apache.org/dyn/closer.cgi/incubator/doris/1.0/1.0.0-incubating/apache-doris-1.0.0-incubating-bin.tar.gz， 未采用源码编译；不知道和操作系统环境是否有关系，还待确定。可以确定的是这个问题是必现。

**Comment by dataalive** [comment]
感谢反馈，这个问题我们正在压测和解决。

**Reply by 980150976** [reply]
补充：有些字段存在大量的null

**Reply by kpfly** [reply]
> 补充：有些字段存在大量的null

Maybe you can try doris 1.1 ，some memory leak issues have been solved.  Thanks. 
https://github.com/apache/doris/releases/tag/1.1.0-rc03
